src/types/name.jl

```diff
@@ -0,0 +1,56 @@
+struct SName
+    region::ShortString15
+    locality1::ShortString15
+    locality2::ShortString15
+end
+
+function Base.print(io::IO, name::SName)
+    print(io, name.region)
+    if !isempty(name.locality1)
+        print(io,"/", name.locality1)
+        if !isempty(name.locality2)
+            print(io,"/", name.locality2)
+        end
+    end
+end
+
+Base.convert(::Type{String}, name::SName) = string(name)
+function Base.convert(::Type{SName}, str::AbstractString)
+    name = try_convert(SName, str)
+    name isa Nothing && DomainError(str, "All timezone name parts must have length < 16")
+    return name
+end
+
+try_convert(::Type{SName}, name::SName) = name
+try_convert(::Type{String}, name::String) = name
+function try_convert(::Type{SName}, str::AbstractString)
+    parts = split(str, "/"; limit=3)
+    all(length(parts) < 16) ||return nothing
+    return if length(parts) == 3
+        SName(parts[1], parts[2], parts[3])
+    elseif length(parts) == 2
+        SName(parts[1], parts[2], ss15"")
+    else
+        SName(parts[1], ss15"", ss15"")
+    end
```

Comment:
Using [incomplete initialization](https://docs.julialang.org/en/v1/manual/constructors/#Incomplete-Initialization) could be cleaner

Reply:
IIRC you can't tell if is isbits field has been initialized if done incompletely.
Since it will be a arbitrary legal value.

Reply:
Yeah you can;'t per the docs on [incomplete initalization](https://docs.julialang.org/en/v1/manual/constructors/#Incomplete-Initialization)

> However, not all object fields are references. Julia considers some types to be "plain data", meaning all of their data is self-contained and does not reference other objects. The plain data types consist of primitive types (e.g. Int) and immutable structs of other plain data types. The initial contents of a plain data type is undefined: